src/server.c

```diff
@@ -4509,13 +4510,20 @@ sds genRedisInfoString(const char *section) {
         }
 
         if (server.loading) {
-            double perc;
+            double perc = 0;
             time_t eta, elapsed;
-            off_t remaining_bytes = server.loading_total_bytes-
-                                    server.loading_loaded_bytes;
-
-            perc = ((double)server.loading_loaded_bytes /
-                   (server.loading_total_bytes+1)) * 100;
+            off_t remaining_bytes = 1;
+
+            if (server.loading_total_bytes) {
+                perc = ((double)server.loading_loaded_bytes / server.loading_total_bytes) * 100;
```

Comment:
@oranagra the original code handled the case of `loading_total_bytes == 0`, how the new code doesn't?

Reply:
@guybe7 you mean the `+1` to avoid division by zero?
this is no longer needed, since we're inside an `if` that checks that it's not zero.

Reply:
oh ok, missed that